### PR TITLE
Fix - #6221 new nav analytics improvements

### LINKED ISSUE #6221: [Enhancement] Improvements in new nav analytics
Need to fix following - 

@akash-codemonk 
- [x] For select widget from suggested widgets list, also add default option
- [x] Add events for navigation to the entities.

@pranavkanade 
- [x] Fix bug in CONNECT DATA - navigates to new datasource even if already have datasource.
- [x] Add generic event for create new datasource.
- [x] Add Widget type/name to CONNECT DATA event.

### PR DESCRIPTION
**Pull request by pranavkanade**
## Description
Adds fixes for the tasks listed following - 
```
 Fix bug in CONNECT DATA - navigates to new datasource even if already have datasource.
 Add generic event for create new datasource.
 Add Widget type/name to CONNECT DATA event.
```

Fixes #6221 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- manually

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/new-nav-analytics-improvements-6221 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 51.48 **(-0.02)** | 33.49 **(-0.02)** | 29.28 **(-0.01)** | 52.11 **(-0.02)**
 :green_circle: | app/client/src/pages/Editor/IntegrationEditor/MockDataSources.tsx | 45.95 **(0.24)** | 87.5 **(0)** | 0 **(0)** | 45.95 **(0.24)**
 :red_circle: | app/client/src/pages/Editor/IntegrationEditor/NewApi.tsx | 34.72 **(-2.04)** | 21.05 **(0)** | 0 **(0)** | 36.76 **(-2.3)**
 :red_circle: | app/client/src/pages/Editor/PropertyPane/ConnectDataCTA.tsx | 84.62 **(-2.22)** | 100 **(0)** | 85.71 **(0)** | 84.21 **(-2.28)**
 :green_circle: | app/client/src/pages/Editor/PropertyPane/index.tsx | 83.19 **(0)** | 72.81 **(0.08)** | 70.37 **(0)** | 84.76 **(0)**
 :red_circle: | app/client/src/utils/helpers.tsx | 53.05 **(-1.83)** | 26.03 **(-4.11)** | 21.43 **(-3.57)** | 46.51 **(-1.55)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.59 **(-0.24)** | 39.41 **(-0.84)** | 36.21 **(0)** | 54.77 **(-0.27)**</details>

### CODE REVIEW

**Review by rishabhsaxena — DISMISSED** [review]

**Review by rishabhsaxena — APPROVED** [review]

File: app/client/src/pages/Editor/IntegrationEditor/NewApi.tsx [hunk 1]
```diff
@@ -211,6 +217,9 @@ function NewApiScreen(props: Props) {
         AnalyticsUtil.logEvent("IMPORT_API_CLICK", {
           importSource: CURL,
         });
+        AnalyticsUtil.logEvent("CREATE_DATA_SOURCE_CLICK", {
```
Comment (Rishabh-Rathod) [inline app/client/src/pages/Editor/IntegrationEditor/NewApi.tsx:220]: Why is `CREATE_DATA_SOURCE_CLICK` analytics inside `API_ACTION.IMPORT_CURL` option? Was it intentional?

File: app/client/src/pages/Editor/IntegrationEditor/MockDataSources.tsx [hunk 2]
```diff
@@ -105,6 +106,10 @@ function MockDatasourceCard(props: MockDatasourceCardProps) {
   }
 
   const addMockDataSource = () => {
+    AnalyticsUtil.logEvent("CREATE_DATA_SOURCE_CLICK", {
```
Comment (Rishabh-Rathod) [inline app/client/src/pages/Editor/IntegrationEditor/MockDataSources.tsx:109]: Q:- Is `CREATE_DATA_SOURCE_CLICK` a general event for all datasource creation? Wouldn't it be better to have more specific events like `CREATE_MOCK_DATASOURCE_CLICK`?

Comment (pranavkanade) [reply app/client/src/pages/Editor/IntegrationEditor/MockDataSources.tsx:109]: This is the requirement. We are also adding a specific analytics for mock data. PR - https://github.com/appsmithorg/appsmith/pull/6136

File: app/client/src/pages/Editor/IntegrationEditor/NewApi.tsx (hunk 1 above)
Comment (pranavkanade) [reply app/client/src/pages/Editor/IntegrationEditor/NewApi.tsx:220]: yes it is. We wanted a single event to track new datasource creation. it's the requirement

**Review by Rishabh-Rathod — APPROVED** [review]

**Comment by pranavkanade** [comment]
Unrelated spec file is failing for all - https://dashboard.cypress.io/projects/eyxvp8/runs/8289/overview

**Pull request merged**